termwiz/src/caps/mod.rs

```diff
@@ -52,74 +52,74 @@
 //! With all this in mind, this module presents a `Capabilities` struct
 //! that holds information about a terminal.   The `new_from_env` method
 //! implements some heuristics (a fancy word for guessing) to compute
-//! the terminal capabilities, but also offers a `ProbeHintsBuilder`
+//! the terminal capabilities, but also offers a `ProbeHints`
 //! that can be used by the embedding application to override those choices.
+use crate::builder;
 use anyhow::Error;
-use derive_builder::*;
 use semver::Version;
 use std::env::var;
 use terminfo::{self, capability as cap};
 
-/// Use the `ProbeHintsBuilder` to configure an instance of
-/// the `ProbeHints` struct.  `ProbeHints` are passed to the `Capabilities`
-/// constructor to influence the effective set of terminal capabilities.
-#[derive(Debug, Default, Builder, Clone)]
-#[builder(default)]
-pub struct ProbeHints {
-    /// The contents of the TERM environment variable
-    term: Option<String>,
+builder! {
+    /// Use the `ProbeHints` to configure an instance of
```

Comment:
Will Rust let us do something like:

```
#[deprecated("Use ProbeHints directly")]
pub type ProbeHintsBuilder = ProbeHints;
```

to help guide anyone that might be using the hint builder?   I don't think this is critical and it may only be streampager that was using this anyway; if we can't deprecate the use of a type alias then we don't need to worry about it.

Reply:
I looked at the generated code and didn't quite like the API:

    pub struct ProbeHintsBuilder {
        #[doc = " The contents of the TERM environment variable"]
        term: ::std::option::Option<Option<String>>,
        ...
    }

    impl ProbeHintsBuilder {
        #[doc = " The contents of the TERM environment variable"]
        #[allow(unused_mut)]
        pub fn term(&mut self, value: Option<String>) -> &mut Self {
            let mut new = self;
            new.term = ::std::option::Option::Some(value);
            new
        }

        pub fn build(&self)
         -> ::std::result::Result<ProbeHints, ::std::string::String> {
             ...
        }
    }

I'd like to get rid of the double `Option`, and make `build` infalliable (or not needed). So the API is incompatible now.

If we want compatibility I can restore the old API.